pyccel/ast/core.py

```diff
@@ -249,7 +251,7 @@ def handle_precedence(args, my_precedence):
 class PyccelBitOperator(Expr, PyccelAstNode):
     _rank = 0
     _shape = ()
-
+    __slots__ = ['_args']
```

Comment:
The `_args` slot should be already defined by the `Expr` class.
See https://stackoverflow.com/questions/1816483/how-does-inheritance-of-slots-in-subclasses-actually-work
for discussions of how `__slots__` works for subclasses

Reply:
By the way, we are going to have a big problem with multiple inheritance.

Reply:
Is there any reason for `PyccelAstNode` to not inherit from `Basic` ? That could help with a lot of these multiple inheritance problems

Reply:
I think `Expr` allows a node to be used in an arithmetic expression, while `Basic` is, well, more "basic"....

Reply:
But since we have the basic arithmetic operators as Pyccel nodes, we should be able to do this. I'm afraid we went down a rabbit hole here :sweat_smile: